CMakeLists.txt

```diff
@@ -40,28 +40,43 @@ IF("${isSystemDir}" STREQUAL "-1")
    SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/wpilib/lib")
 ENDIF("${isSystemDir}" STREQUAL "-1")
 
-option(WITHOUT_JAVA "don't include java and JNI in the build" OFF)
-option(BUILD_SHARED_LIBS "build with shared libs (needed for JNI)" ON)
-option(WITHOUT_CSCORE "Don't build cscore (removes OpenCV requirement)" OFF)
-option(WITHOUT_ALLWPILIB "Don't build allwpilib (removes OpenCV requirement)" ON)
-option(WITH_TESTS "build unit tests (requires internet connection)" OFF)
-option(USE_EXTERNAL_HAL "Use a separately built HAL" OFF)
+# Options for building certain parts of the repo. Everything is built by default.
+option(WITH_JAVA "Include java and JNI in the build" ON)
+option(WITH_SHARED_LIBS "Build with shared libs (needed for JNI)" ON)
+option(WITH_CSCORE "Build cscore (needs OpenCV)" ON)
+option(WITH_ALLWPILIB "Build allwpilib (needs OpenCV)" ON)
+option(WITH_TESTS "Build unit tests (requires internet connection)" ON)
+option(WITH_SIMULATION_MODULES "Build simulation modules" ON)
+
+# Options for external HAL.
+option(WITH_EXTERNAL_HAL "Use a separately built HAL" OFF)
 set(EXTERNAL_HAL_FILE "" CACHE FILEPATH "Location to look for an external HAL CMake File")
+
+# Options for using a package manager (vcpkg) for certain dependencies.
 option(USE_VCPKG_LIBUV "Use vcpkg libuv" OFF)
 option(USE_VCPKG_EIGEN "Use vcpkg eigen" OFF)
-option(FLAT_INSTALL_WPILIB "Use a flat install directory" OFF)
-option(WITH_SIMULATION_MODULES "build simulation modules" OFF)
+
+# Options for installation.
+option(WITH_FLAT_INSTALL_WPILIB "Use a flat install directory" OFF)
```

Comment:
Can probably remove the _WPILIB suffix on this option?  Unless it only affects wpilibc/j?